src/Evolution/Systems/RelativisticEuler/Valencia/TimeDerivative.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace {
+template <size_t Dim>
+void densitized_stress(
+    const gsl::not_null<tnsr::II<DataVector, Dim, Frame::Inertial>*> result,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& tilde_s_vector,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& spatial_velocity,
+    const tnsr::II<DataVector, Dim, Frame::Inertial>& inv_spatial_metric,
+    const Scalar<DataVector>& sqrt_det_spatial_metric,
+    const Scalar<DataVector>& pressure) noexcept {
+  *result = inv_spatial_metric;
+  for (size_t i = 0; i < Dim; ++i) {
+    for (size_t j = i; j < Dim; ++j) {
+      result->get(i, j) *= get(sqrt_det_spatial_metric) * get(pressure);
+      result->get(i, j) +=
+          0.5 * (tilde_s_vector.get(i) * spatial_velocity.get(j) +
+                 tilde_s_vector.get(j) * spatial_velocity.get(i));
+    }
+  }
+}
+}  // namespace
+
+namespace RelativisticEuler::Valencia {
+namespace detail {
+template <size_t Dim>
+void fluxes_impl(
+    gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> tilde_d_flux,
+    gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> tilde_tau_flux,
+    gsl::not_null<tnsr::Ij<DataVector, Dim, Frame::Inertial>*> tilde_s_flux,
+    gsl::not_null<DataVector*> transport_velocity_I,
+    const Scalar<DataVector>& tilde_d, const Scalar<DataVector>& tilde_tau,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& tilde_s,
+    const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& shift,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& spatial_velocity,
+    const DataVector& p_alpha_sqrt_det_g) noexcept;
+
+template <size_t Dim>
+void sources_impl(
+    gsl::not_null<Scalar<DataVector>*> source_tilde_tau,
+    gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*> source_tilde_s,
+    gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*> tilde_s_M,
+    gsl::not_null<tnsr::II<DataVector, Dim, Frame::Inertial>*> tilde_s_MN,
+    const Scalar<DataVector>& tilde_d, const Scalar<DataVector>& tilde_tau,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& tilde_s,
+    const tnsr::I<DataVector, Dim, Frame::Inertial>& spatial_velocity,
+    const Scalar<DataVector>& pressure, const Scalar<DataVector>& lapse,
+    const tnsr::i<DataVector, Dim, Frame::Inertial>& d_lapse,
+    const tnsr::iJ<DataVector, Dim, Frame::Inertial>& d_shift,
+    const tnsr::ijj<DataVector, Dim, Frame::Inertial>& d_spatial_metric,
+    const tnsr::II<DataVector, Dim, Frame::Inertial>& inv_spatial_metric,
+    const Scalar<DataVector>& sqrt_det_spatial_metric,
+    const tnsr::ii<DataVector, Dim, Frame::Inertial>&
+        extrinsic_curvature) noexcept;
+}  // namespace detail
+
+template <size_t Dim>
+void TimeDerivative<Dim>::apply(
+    const gsl::not_null<Scalar<DataVector>*> dt_tilde_d,
+    const gsl::not_null<Scalar<DataVector>*> dt_tilde_tau,
+    const gsl::not_null<tnsr::i<DataVector, Dim, Frame::Inertial>*> dt_tilde_s,
```

Comment:
How about `dt_non_flux_terms_tilde_d`? A bit long, but it's the source and non-conservative product terms that should be added.

Reply:
I like 'non flux terms', that's a good suggestion! What would you think of permuting the words to `dt_tilde_d_non_flux_terms` (or `non_flux_terms_dt_tilde_d`)? To me, at least, that's clearer that it's some terms for the d/dt, rather than the d/dt of some terms.

Reply:
Hmm, I'm fine with either. I can't decide which I like better. Do you have a preference?

Reply:
I'll voice a slight preference for `dt_tilde_d_non_flux_terms`, but I'd be happy with any of these (your suggestion and my two variants). Feel free to summon Larry if you'd like his opinion before making any changes :)

Reply:
@kidder prefers `non_flux_terms_dt_tilde_d` because it's the non-flux terms of `dt(variable)` :shrug: I'll go with that. I think any of them are a big improvement :D

Reply:
Sounds good to me!